### PR TITLE
Added initial configurations for project

### PR DESCRIPTION
**Pull request by Vaserty**
This PR adding initialize config

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+func main() {
+	logger.Info("Initializing application...")
+	injector := do.New()
+	environment := config.GetEnvironment()
+	logger.Infof("Running in environment: '%v'", environment)
+	buildedConfig, err := provideConfig(injector, environment)
+	if err != nil {
+		logger.Fatalf("Failed to initialize configuration: %v", err)
```
Comment (imatw4r) [inline cmd/main.go]: Why are you not panicing if you are not able to initialzie config?

File: internal/config/logger.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	logger "github.com/sirupsen/logrus"
+)
+
+func ConfigLogger(config *Config) error {
+	level, err := logger.ParseLevel(config.Logger.Level)
+	if err != nil {
+		return fmt.Errorf("error parsing log level: %w", err)
+	}
+	logger.SetLevel(level)
+	logger.Info("Logger configured successfully.")
```
Comment (imatw4r) [inline internal/config/logger.go]: I always wonder: Shouldn't this log be in main.go after logger has been initialized? It's like if I was to use this in my application, there was no way to disable this log.

File: internal/customtypes/config_types.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package customtypes
+
+type Environment string
+
+const (
+	Development Environment = "development"
```
Comment (imatw4r) [inline internal/customtypes/config_types.go]: I'd suggest renaming to `EnvDev/EnvProd/EnvTest= ...` so when you import it's much easier to use and reason:

```go
import github.com/Vaserty/go-authorization/internal/customtypes

x := Development
```
vs 
```go
import github.com/Vaserty/go-authorization/internal/customtypes

x := EnvDev
```

**Review by imatw4r — APPROVED** [review]

File: internal/customtypes/config_types.go (hunk 3 above)
Comment (Vaserty) [reply internal/customtypes/config_types.go]: Yes, You're right,

File: internal/config/logger.go (hunk 2 above)
Comment (Vaserty) [reply internal/config/logger.go]: Hmm, you're actually right, I would also expect this log in the main.go file

**Pull request merged**